src/object/srv_ec_aggregate.c

```diff
@@ -894,44 +806,24 @@ agg_update_vos(struct ec_agg_param *agg_param, struct ec_agg_entry *entry,
 			daos_csummer_free_ic(csummer, &iod_csums);
 		if (rc) {
 			D_ERROR("vos_obj_update failed: "DF_RC"\n", DP_RC(rc));
-			goto out;
+			D_GOTO(out, rc);
 		}
 	}
 
-	if (entry->ae_cur_stripe.as_ho_ext_cnt == 0 && agg_contained(entry)) {
-		epoch_range = ap->ap_epr;
-		recx.rx_nr = ec_age2ss(entry);
-		recx.rx_idx = entry->ae_cur_stripe.as_stripenum * recx.rx_nr;
-		rc = vos_obj_array_remove(ap->ap_cont_handle,
-					  entry->ae_oid, &epoch_range,
-					  &entry->ae_dkey, &entry->ae_akey,
-					  &recx);
+	d_list_for_each_entry(ext, &entry->ae_cur_stripe.as_dextents, ae_link) {
+		int err;
 
-	} else {
-		d_list_for_each_entry(ext, &entry->ae_cur_stripe.as_dextents,
-				      ae_link) {
-			int		erc = 0;
-			uint64_t	se;
-
-			se = ec_age2ss(entry) *
-			     (entry->ae_cur_stripe.as_stripenum + 1);
-			if (DAOS_RECX_END(ext->ae_orig_recx) <= se) {
-				epoch_range.epr_lo = epoch_range.epr_hi =
-					ext->ae_epoch;
-
-				erc = vos_obj_array_remove(ap->ap_cont_handle,
-							   entry->ae_oid,
-							   &epoch_range,
-							   &entry->ae_dkey,
-							   &entry->ae_akey,
-							   &ext->ae_orig_recx);
-				if (erc)
-					D_ERROR("array_remove fails: "DF_RC"\n",
-						DP_RC(erc));
-				if (!rc && erc)
-					rc = erc;
-			}
-		}
+		epoch_range.epr_lo = epoch_range.epr_hi = ext->ae_epoch;
+		err = vos_obj_array_remove(ap->ap_cont_handle,
+					   entry->ae_oid,
+					   &epoch_range,
+					   &entry->ae_dkey,
+					   &entry->ae_akey,
+					   &ext->ae_recx);
```

Comment:
probably need not traverse exts in as_dextents, can just vos_obj_array_remove() this full-stripe on this parity shard? (similar as the _remove in ds_obj_ec_agg_handler).